gap/AutoDocMainFunction.gi

```diff
@@ -419,3 +419,65 @@ InstallGlobalFunction( CreateMakeTest,
 
     CloseStream( filestream );
 end );
+
+# The following function is originally by Alexander Konovalov
+BindGlobal("AUTODOC_ExtractMyManualExamples",
+function( pkgname, pkgdir, docdir, main, files )
+    local tst, i, s, basename, name, output, ch, a, location, pos, comment, pkgdirString;
+    Print("Extracting manual examples for ", pkgname, " package ...\n" );
+    if not EndsWith(main, ".xml") then
+        main := Concatenation( main, ".xml" );
+    fi;
+    tst:=ExtractExamples( docdir, main, files, "Chapter" );
+    Print(Length(tst), " chapters detected\n");
+    pkgdirString := Filename(pkgdir, "");
+    AUTODOC_CreateDirIfMissing( Filename(pkgdir, "tst") );
+    for i in [ 1 .. Length(tst) ] do
+        Print( "Chapter ", i, " : \c" );
+        if Length( tst[i] ) = 0 then
+            Print("no examples \n" );
+            continue;
+        fi;
+        s := String(i);
+        if Length(s)=1 then
+            # works for <100 chapters
+            s:=Concatenation("0",s);
+        fi;
+        basename := Concatenation( LowercaseString(pkgname), s, ".tst" );
+        basename := ReplacedString( basename, " ", "_" );
+        name := Filename(pkgdir, Concatenation("tst/", basename) );
+        output := OutputTextFile( name, false ); # to empty the file first
+        SetPrintFormattingStatus( output, false ); # to avoid line breaks
+        ch := tst[i];
+        AppendTo(output, "# ", pkgname, ", chapter ",i,"\n");
+        AppendTo(output, "#\n",
+                         "# DO NOT EDIT THIS FILE - EDIT EXAMPLES IN THE SOURCE INSTEAD!\n",
+                         "#\n",
+                         "# This file has been autogenerated with GAP. It contains examples\n",
+                         "# extracted from the documentation. Each example is preceded by the\n",
+                         "# comment which points to the location of its source.\n",
```

Comment:
I am not quite sure about adding the source here.

In most AutoDoc cases, the source will be a generated `.xml` file, so I am not sure how useful it would be. It might even irritate people not aware of the fact that AutoDoc generates XML-files.

Reply:
Having the origin of each test is super useful for people who just use AutoDoc to convert GAPDoc XML inputs. Of course it is less useful if one uses full-blown AutoDoc, including for the inputs; but I'd argue even then it can be helpful, as by looking at the XML, I can usually at least identify some context 

Anyway, I wouldn't mind if we only inserted the "source location" for non-AutoDoc-XML files (to keep things simple, that would mean excluding sources starting with an `_`). But is it really worth the effort to strip this information? I don't quite understand what is irritating about it ... at least if we were to adjust the comment to clarify that if the XML "source" is generated, then of course it is not the *actual* source...

Reply:
Adjusting the comment would be sufficient, yes. I just assume it might be irritating for people who have not really seen the xml files before.

Reply:
@sebasguts I've update this text now. Better?